storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/MetricCollector.java

```diff
@@ -55,41 +58,67 @@ class MetricCollector implements MetricPublisher {
+        requestMetrics.put("GetObject", createRequestsSensor("get-object-requests"));
+        latencyMetrics.put("GetObject", createLatencySensor("get-object-time"));
+        requestMetrics.put("UploadPart", createRequestsSensor("upload-part-requests"));
+        latencyMetrics.put("UploadPart", createLatencySensor("upload-part-time"));
+        requestMetrics.put("CreateMultipartUpload", createRequestsSensor("create-multipart-upload-requests"));
+        latencyMetrics.put("CreateMultipartUpload", createLatencySensor("create-multipart-upload-time"));
+        requestMetrics.put("CompleteMultipartUpload", createRequestsSensor("complete-multipart-upload-requests"));
+        latencyMetrics.put("CompleteMultipartUpload", createLatencySensor("complete-multipart-upload-time"));
+        requestMetrics.put("PutObject", createRequestsSensor("put-object-requests"));
+        latencyMetrics.put("PutObject", createLatencySensor("put-object-time"));
+        requestMetrics.put("DeleteObject", createRequestsSensor("delete-object-requests"));
+        latencyMetrics.put("DeleteObject", createLatencySensor("delete-object-time"));
+        requestMetrics.put("AbortMultipartUpload", createRequestsSensor("abort-multipart-upload-requests"));
+        latencyMetrics.put("AbortMultipartUpload", createLatencySensor("abort-multipart-upload-time"));
+
+        errorMetrics.put(THROTTLING.toString(), createRequestsSensor("throttling-errors"));
+        errorMetrics.put(SERVER_ERROR.toString(), createRequestsSensor("server-errors"));
+        errorMetrics.put(CONFIGURED_TIMEOUT.toString(), createRequestsSensor("configured-timeout-errors"));
+        errorMetrics.put(IO.toString(), createRequestsSensor("io-errors"));
+        errorMetrics.put(OTHER.toString(), createRequestsSensor("other-errors"));
     }
 
-    private Sensor createSensor(final String name) {
+    private Sensor createRequestsSensor(final String name) {
         final Sensor sensor = metrics.sensor(name);
         sensor.add(metrics.metricName(name + "-rate", METRIC_GROUP), new Rate());
         sensor.add(metrics.metricName(name + "-total", METRIC_GROUP), new CumulativeCount());
         return sensor;
     }
 
+    private Sensor createLatencySensor(final String name) {
+        final Sensor sensor = metrics.sensor(name);
+        sensor.add(metrics.metricName(name + "-max", METRIC_GROUP), new Max());
+        sensor.add(metrics.metricName(name + "-avg", METRIC_GROUP), new Avg());
+        return sensor;
+    }
+
     @Override
     public void publish(final MetricCollection metricCollection) {
         final List<String> metricValues = metricCollection.metricValues(CoreMetric.OPERATION_NAME);
-        for (final String metricValue : metricValues) {
-            final var sensor = requestMetrics.get(metricValue);
-            if (sensor != null) {
-                sensor.record();
+        // metrics are reported per request, so 1 value can be assumed.
+        if (metricValues.size() == 1) {
```

Comment:
The for loop was introduced intentionally after the discussion with @ivanyu and I guess it will not affect anything if what you specified in the comment is true, however both me and @ivanyu were not sure if that's the case. So could you elaborate how did you find out that the size will always be 1?

Reply:
API is not clear about it, but docs mention the following: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/metrics-list.html 

> Metrics collected with each request

So, my interpretation (and while testing this) is that each instance of metric collection corresponds to a single API call.

Reply:
Ok, I looked into the SDK code and seems you are right :+1:

Reply:
But OTOH what harm would it do it it remains a loop? It'd be more future-proof and we wouldn't expect this internal behavior to change in a casual dependency upgrade

Reply:
The thing is that we will not be able to understand to what request type the latency metric belongs if there will be multiple values AFAIU. Otherwise I would agree.

Reply:
Oh, I see. That's very unfortunate. Well, let's rely on our tests as much as we can then